pkg/utils/istio/virtualservice.go

```diff
@@ -35,14 +35,33 @@ func VirtualServiceWithSNIMatch(virtualService *istionetworkingv1beta1.VirtualSe
 }
 
 // VirtualServiceForTLSTermination returns a function for use with a gateway that performs TLS termination.
-func VirtualServiceForTLSTermination(virtualService *istionetworkingv1beta1.VirtualService, labels map[string]string, hosts []string, gatewayName string, port uint32, destinationHost string) func() error {
+func VirtualServiceForTLSTermination(virtualService *istionetworkingv1beta1.VirtualService, labels map[string]string, hosts []string, gatewayName string, port uint32, destinationHost, destinationUpgradeHost, connectionUpgradeRouteName string) func() error {
 	return func() error {
 		virtualService.Labels = labels
 		virtualService.Spec = istioapinetworkingv1beta1.VirtualService{
 			ExportTo: []string{"*"},
 			Hosts:    hosts,
 			Gateways: []string{gatewayName},
 			Http: []*istioapinetworkingv1beta1.HTTPRoute{
+				{
+					Name: connectionUpgradeRouteName,
```

Comment:
Is this name required? In the existing `HTTPRoute` below it is not specified. Not sure if we need the name unless envoy complains having two `HTTPRoute`s without names.

Reply:
Yes, it is required. The route is updated by an `EnvoyFilter` and we need something to match ([ref](https://github.com/gardener/gardener/blob/b6649a8d5aff062e9a312626225d285428d2cbe2/pkg/component/kubernetes/apiserverexposure/templates/envoyfilter-istio-tls-termination.yaml#L78)).
The default route is not updated so it does not need a name.